index.js

```diff
@@ -48,6 +49,12 @@ export default class Carousel extends Component {
         }
     }
 
+    componentDidUpdate (prevProps) {
+        if (this.props.children && prevProps.children && this.props.children.length !== prevProps.children.length) {
+            this.goToPage(this.props.initialPage);
```

Comment:
I have a variable number of cards so if the number of children has changed I allow the parent component to set the initial page again. I figured this was a bit of a specific use case PR so feel free to decline :-)